modules/apps/frontend-js/frontend-js-a11y-web/src/main/resources/META-INF/resources/Violations.tsx

```diff
@@ -0,0 +1,255 @@
+								selectedCategories.findIndex(
+									(element) => element === item.value
+								) !== -1;
+
+							return (
+								<ClayDropDown.Section key={item.value}>
+									<ClayCheckbox
+										checked={existsCategory}
+										data-testid={item.value}
+										label={item.label}
+										onChange={() => {
+											if (existsCategory) {
+												dispatch({
+													payload: {
+														value: item.value,
+													},
+													type: TYPES.CATEGORY_REMOVE,
+												});
+
+												return;
+											}
+											dispatch({
+												payload: {
+													value: item.value,
+												},
+												type: TYPES.CATEGORY_ADD,
+											});
+										}}
+									></ClayCheckbox>
+								</ClayDropDown.Section>
+							);
+						})}
+					</ClayDropDown.Group>
+				</ClayDropDown.ItemList>
+			</ClayDropDown>
+		</>
+	);
+}
+
+type TViolationNext = {
+	violationIndex: number;
+};
+
+type ViolationsProps = {
+	next?: (payload: TViolationNext) => void;
+	selectedCategories: Array<String>;
+	selectedImpact: Array<ImpactValue>;
+	violations: Array<Result>;
+};
+
+export default function Violations({
+	next,
+	selectedCategories,
+	selectedImpact,
+	violations,
+}: ViolationsProps) {
+	const hasViolations = !!violations.length;
+
+	return (
+		<>
+			<div className="sidebar-section">
+				<div className="page-accessibility-tool__sidebar--violations-panel-header-title">
+					<div className="inline-item inline-item-before">
+						<ClayIcon
+							className="text-danger"
+							symbol="info-circle"
+						/>
+					</div>
+					<div className="inline-item">
+						<span className="list-group-title">
+							{Liferay.Language.get('accessibility-violations')}
+						</span>
+					</div>
+					<div className="inline-item inline-item-after">
+						<ViolationsFilter
+							selectedCategories={selectedCategories}
+							selectedImpact={selectedImpact}
+						/>
+					</div>
+				</div>
+			</div>
+			<div className="page-accessibility-tool__sidebar--violations-panel-header-description">
+				{!hasViolations
+					? Liferay.Language.get(
+							'there-are-no-accessibility-violations-in-this-page'
+					  )
+					: Liferay.Language.get(
+							'set-of-rules-violated-by-the-highlighted-issues'
+					  )}
+			</div>
+			{hasViolations && (
+				<ClayList
+					aria-label={Liferay.Language.get('violations-list')}
+					className="list-group-flush"
+					role="tablist"
+				>
+					{violations.map((violation, index) => {
+						const {id, impact, nodes} = violation;
+
+						return (
+							<Rule
+								aria-label={Liferay.Util.sub(
+									Liferay.Language.get(
+										'navigate-to-violation-x'
+									),
+									id
+								)}
+								key={id}
+								onClick={() => {
+									if (next) {
+										next({violationIndex: index});
+									}
+								}}
+								quantity={nodes.length}
+								ruleSubtext={
+									impact as string | React.ReactChildren
```

Comment:
Can `impact` be a `React.ReactChildren`? I'm a little confused.

Reply:
I changed to `React.ReactNode` type and it worked well. [`React.ReactNode` looks the more generic React type :)](https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/basic_type_example/#useful-react-prop-type-examples)

Reply:
Yes, but impact will be a string and not a `React.ReactNode` does not make sense here.

Reply:
From React.ReactNode to ImpactValue and optional